src/Bridge/Symfony/Validator/Metadata/Property/ValidatorPropertyMetadataFactory.php

```diff
@@ -48,26 +79,38 @@ public function __construct(ValidatorMetadataFactoryInterface $validatorMetadata
     public function create(string $resourceClass, string $name, array $options = []): PropertyMetadata
     {
         $propertyMetadata = $this->decorated->create($resourceClass, $name, $options);
-        if (null !== $propertyMetadata->isRequired()) {
+
+        $required = $propertyMetadata->isRequired();
+        $iri = $propertyMetadata->getIri();
+
+        if (null !== $required && null !== $iri) {
             return $propertyMetadata;
         }
 
         $validatorClassMetadata = $this->validatorMetadataFactory->getMetadataFor($resourceClass);
         foreach ($validatorClassMetadata->getPropertyMetadata($name) as $validatorPropertyMetadata) {
-            if (isset($options['validation_groups'])) {
-                return $propertyMetadata->withRequired($this->isRequiredByGroups($validatorPropertyMetadata, $options));
+            if (null === $required && isset($options['validation_groups'])) {
+                $required = $this->isRequiredByGroups($validatorPropertyMetadata, $options);
             }
 
             foreach ($validatorPropertyMetadata->findConstraints($validatorClassMetadata->getDefaultGroup()) as $constraint) {
-                if ($this->isRequired($constraint)) {
-                    return $propertyMetadata->withRequired(true);
+                if (null === $required && $this->isRequired($constraint)) {
+                    $required = true;
                 }
-            }
 
-            return $propertyMetadata->withRequired(false);
+                if (null === $iri) {
+                    $iri = self::SCHEMA_MAPPED_CONSTRAINTS[\get_class($constraint)] ?? null;
+                }
+
+                if (null !== $required && null !== $iri) {
+                    break 2;
+                }
+            }
         }
 
-        return $propertyMetadata->withRequired(false);
+        $propertyMetadata = $propertyMetadata->withIri($iri);
+
+        return $propertyMetadata->withRequired($required ?? false);
```

Comment:
wdyt about `$propertyMetadata->withIri($iri)->withRequired($required ?? false);` instead of assigning the variable?

Reply:
yup sure! This is actually better :)